src/xrGame/cover_manager.cpp

```diff
@@ -74,37 +74,42 @@ void CCoverManager::compute_static_cover()
 {
     clear();
     xr_delete(m_covers);
-    m_covers = xr_new<CPointQuadTree>(
-        ai().level_graph().header().box(), ai().level_graph().header().cell_size() * .5f, 8 * 65536, 4 * 65536);
-    m_temp.resize(ai().level_graph().header().vertex_count());
-
+    
     const CLevelGraph& graph = ai().level_graph();
-    const u32 levelVertexCount = ai().level_graph().header().vertex_count();
+    const LevelGraph::CHeader &levelGraphHeader = graph.header();
+    const u32 levelVertexCount = levelGraphHeader.vertex_count();
+
+    m_covers = xr_new<CPointQuadTree>(levelGraphHeader.box(), levelGraphHeader.cell_size() * .5f, 8 * 65536, 4 * 65536);
 
+    // avoiding extra allocations with a static storage for m_covers
+    static xr_vector<std::optional<CCoverPoint>> quadTreeStaticStorage;
```

Comment:
I'm not sure if this is a good trade-off between alocs/RSS

Reply:
It's not recommended to use global/static objects which allocate memory dynamically.